src/overlays/actors/ovl_Bg_Ladder/z_bg_ladder.c

```diff
@@ -19,22 +37,129 @@ const ActorInit Bg_Ladder_InitVars = {
+        this->dyna.actor.draw = NULL;
+        this->action = BgLadder_ActionWait;
+    }
+}
+
+void BgLadder_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgLadder* this = THIS;
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void BgLadder_ActionWait(BgLadder* this, GlobalContext* globalCtx) {
+    // Wait for the flag to be set, then trigger the cutscene
+    if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+        this->action = BgLadder_ActionStartCutscene;
+    }
+}
+
+void BgLadder_ActionStartCutscene(BgLadder* this, GlobalContext* globalCtx) {
+    // Trigger the cutscene, then make the ladder fade in
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        this->dyna.actor.draw = BgLadder_Draw;
+        Audio_PlayActorSound2(this, 0x2997);
+        this->action = BgLadder_ActionFadeIn;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
+
+void BgLadder_ActionFadeIn(BgLadder* this, GlobalContext* globalCtx) {
+    // Fade in the ladder, then stop the cutscene & go idle
+    this->alpha += 5;
+    if (this->alpha >= 255) {
+        this->alpha = 255;
+        ActorCutscene_Stop(this->dyna.actor.cutscene);
+        func_800C6314(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        this->dyna.actor.flags &= ~0x10; // always update = off
+        this->action = BgLadder_ActionIdle;
+    }
+}
+
+void BgLadder_ActionIdle(BgLadder* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Ladder_0x80A4D9F0/BgLadder_Destroy.asm")
+void BgLadder_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgLadder* this = THIS;
+    this->action(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Ladder_0x80A4D9F0/func_80A4DB88.asm")
+void BgLadder_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgLadder* this = THIS;
+    s32 pad;
+    Gfx* gfx;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Ladder_0x80A4D9F0/func_80A4DBD8.asm")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Ladder_0x80A4D9F0/func_80A4DC48.asm")
+    if (this->alpha == 255) {
+        Scene_SetRenderModeXlu(globalCtx, 0, 0x01);
+        gfx = POLY_OPA_DISP;
+    } else {
+        Scene_SetRenderModeXlu(globalCtx, 1, 0x02);
+        gfx = POLY_XLU_DISP;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Ladder_0x80A4D9F0/func_80A4DCCC.asm")
+    gSPDisplayList(&gfx[0], &sSetupDL[6 * 0x19]);
+    gDPSetEnvColor(&gfx[1], 255, 255, 255, this->alpha);
+    gSPMatrix(&gfx[2], Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(&gfx[3], sLadderDLists[this->dyna.actor.params]);
```

Comment:
Does it still match if you try using `gfx++` instead of indexing, then set `POLY_OPA_DISP` or `POLY_XLU_DISP` to gfx at the end?